src/PrestaShopBundle/Service/TranslationService.php

```diff
@@ -181,7 +191,7 @@ public function listDomainTranslation($locale, $domain, $theme = null, $search =
             $defaultCatalog = current($translationProvider->getDefaultCatalogue()->all());
         }
 
-        $xliffCatalog = current($translationProvider->getXliffCatalogue()->all());
+        $xliffCatalog = method_exists($translationProvider, 'getLegacyCatalogue') ? $translationProvider->getLegacyCatalogue()->all($domain) : $translationProvider->getXliffCatalogue()->all();
```

Comment:
You remove the current function, is it still ok?

Reply:
I guess yes, as all tests pass and I've personally checked the rendering for FO, BO, 3 modules, Others and Mails in the Translations Manager.

I'm not quite sure at this point why `current` was needed: this is what I hate when people rely on native PHP functions instead of creating their own to apply business logic :/

Reply:
Should I remove the previous one too, or should I introduce it back the missing one?

Reply:
No idea if it was needed or not. But if it works without, don't add it :smile: